### DISCUSSION TITLE
社交还是记录

### DISCUSSION THREAD

**Discussion by woodwm** [kingwrcy/mblog-backend#7] [General]
目前这个项目是有点混乱，因为支持多用户，所以是一个自建的微博网站，类似微博，Twitter。因为不支持ActivityPub，所以相比成熟的mastodon 来说没什么优势。现在的项目离flomo 和 memos 已经很远，虽然看起来最初的想法只是加上评论的功能。

如果还是沿着个人记录的方向，因为需要考虑每个帖子是不是公开，无形中增加了负担。我更愿意接受的是以私人记录为核心，也就是默认所有帖子都是不公开的。但能够单独设置部分帖子为公开，类似 Notion 的思路。当然最左侧的内容也是需要修改的，因为我不需要一直关心广场的内容。

如果就是为了做单人的博客平台，要考虑的是目前成本比较低的博客方式，是用静态网页加评论插件。非要自己托管的话，也有WordPress 等可选，很难有动力去用这个早期的项目。

多人博客的话，就是我最上面说的，相比 mastodon 没有优势。当然可以说需要的资源比较小是一个优点。但既然决定自建给多人使用，稍微大点的资源消耗也是可以接受的，而且 mastodon 有很多手机app 可以用，很方便。可以follow 其他服务器的更多的人也是 Fediverse 的吸引人的地方。在自建多人微博的方向上，很难跟mastodon 去竞争，人家起步早，而且现在特别火，得到全世界的认可。

我的建议是沿着个人记录的方向走，默认私有，同时保留一定的社交，但不要打扰私密记录。

**Comment by kingwrcy** [comment]
感谢对mblog的使用与反馈.

mblog的初衷是个人使用的`微`博客.

所谓`微`就是不用长篇大论的像传统博客一样的写文章,那样感觉很有压力,时常没有什么发的.

开发的过程中一直接受网友的反馈,最大的问题就是希望有评论.

memos默认没有评论,很多第三方的方案,看到大把的人各种折腾.

于是我加上了评论的功能,加上了评论之后发现是不是可以收藏别人的发言,于是又有了`点赞`,相当于收藏.

做到这里我也发觉偏离了初心,出现了多人版,于是又在后台加上了一系列的开关,可以关闭`评论/点赞`.

也可以关闭`多用户`版本.同时也加入了匿名评论+审核的方式,这样可以完全摒弃掉多用户版,只保留匿名评论.

至于`广场`功能,单用户的版本则不可见.

另外`默认可见性`是可以在后台->用户设置里设置的,不用每次都选,默认是公开,可以设置为默认自己可见.默认的是否可评论,也是可以后台设置的.

最后,目前项目还是在初期,才开源一个多月,多用户版的功能基本不再开发,只维护现有的bug,主要精力还是在于服务于单用户.

最近也开发了[Chrome浏览器插件](https://github.com/mblog-backend/browser-plugin),目前很简陋,但是大部分功能都有了.

也感谢[网友@cooolr](https://github.com/cooolr)开发了[自部署的tg机器人](https://github.com/mblog-backend/mblog-telegram-bot),也欢迎大家尝试.

下一阶段的主要功能开发:

1. 提供各种第三方服务的导入工具,首期提供 memos 和 flomo.
